src/main/java/seedu/address/ui/NoteWindow.java

```diff
@@ -0,0 +1,184 @@
+package seedu.address.ui;
+
+import java.util.ArrayList;
+
+import javafx.collections.ObservableList;
+import javafx.fxml.FXML;
+import javafx.scene.control.TextArea;
+import javafx.scene.input.KeyCode;
+import javafx.scene.input.KeyCodeCombination;
+import javafx.scene.input.KeyCombination;
+import javafx.scene.input.KeyCombination.Modifier;
+import javafx.scene.input.KeyEvent;
+import javafx.stage.Stage;
+import seedu.address.commons.util.DateUtil;
+import seedu.address.logic.Logic;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.person.Note;
+import seedu.address.model.person.NoteDate;
+import seedu.address.model.person.Person;
+
+
+
+public class NoteWindow extends UiPart<Stage> {
+
+    protected static final ArrayList<NoteWindow> OPENED_NOTE_WINDOWS = new ArrayList<>();
+    private static final String FXML = "NoteWindow.fxml";
+    private static final Modifier CTRL = KeyCombination.SHORTCUT_DOWN;
+    private static final KeyCombination SAVE_KEY = new KeyCodeCombination(KeyCode.S, CTRL);
+    private static final KeyCombination EXIT_AND_SAVE_KEY = new KeyCodeCombination(KeyCode.Q, CTRL);
+    private static final KeyCombination EXIT_KEY = new KeyCodeCombination(KeyCode.W, CTRL);
+    private static final KeyCombination TIME_STAMP_KEY = new KeyCodeCombination(KeyCode.T, CTRL);
+
+    @FXML
+    private TextArea noteTextArea;
+
+    private final Person person;
+
+    private final Logic logic;
+
+
+    /**
+     * Creates a new NoteWindow.
+     */
+    public NoteWindow(Person person, Logic logic) {
+        super(FXML);
+        noteTextArea.setText(person.getNote().value);
+        this.person = person;
+        this.logic = logic;
+        getRoot().setTitle(person.getName().toString());
+    }
+
+    /**
+     * Show the noteWindow.
+     */
+    public void show() {
+        getRoot().show();
+        getRoot().centerOnScreen();
+    }
+
+    /**
+     * Returns true if the note window is currently being shown.
+     */
+    public boolean isShowing() {
+        return getRoot().isShowing();
+    }
+
+    /**
+     * Hides the note window.
+     */
+    public void hide() {
+        getRoot().hide();
+    }
+
+    /**
+     * Focuses on the note window.
+     */
+    public void focus() {
+        getRoot().setIconified(false);
+        getRoot().requestFocus();
+    }
+
+    /**
+     * Saves the file
+     */
+    @FXML
+    public void handleSave() throws CommandException {
+        String paragraph = noteTextArea.getText();
+        Note editedNote = new Note(paragraph);
+        NoteDate newNoteDate = new NoteDate(DateUtil.getCurrentDateTime());
+        Person editedPerson = new Person(person.getName(), person.getPhone(), person.getEmail(),
+                editedNote, newNoteDate, person.getTags());
+        logic.executeSaveNote(person, editedPerson);
+    }
+
+
+    /**
+     * Exits the note window.
+     */
+    @FXML
```

Comment:
I think it would be good to introduce confirmation -- It's not difficult to type command w instead of command q, and if you didn't save... that's not great. And it's also not hard to type command w instead of command s.

Either that, or a more troublesome command to close without saving?